File: lib-model/src/main/java/com/minesec/msa/client/sdk/management/model/ActivationResponseDto.java
package com.minesec.msa.client.sdk.management.model;

import com.minesec.msa.client.sdk.ResultDto;
import com.minesec.msa.client.sdk.TMSErrorCode;

import lombok.Getter;

/**
 * @author eric.song
 * @since 2023/4/20 10:25
 */
@Getter
public class ActivationResponseDto extends ResultDto {

    private ActivationData data;

    @Override
    public boolean approved() {
        // Invalid Activation code -10013, App need to display this error info differently.
        return getRespCode().equals("0") || getRespCode().equals(TMSErrorCode.INVALID_ACTIVATION_CODE);
    }
}

File: lib-model/src/main/java/com/minesec/msa/client/sdk/management/model_k/ActivationResponseDto_.kt
package com.minesec.msa.client.sdk.management.model_k


/**
 * @author eric.song
 * @since 2024/6/22 23:28
 */
data class ActivationResponseDto_(val data: ActivationData_, val success: Int, val message: String): ResultDto_(success, message)
